packages/js/components/src/split-dropdown/split-dropdown.tsx

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import { cloneElement, createElement } from '@wordpress/element';
+import { Icon, chevronDown, chevronUp } from '@wordpress/icons';
+import { Button, ButtonGroup, Dropdown } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+export type SplitDropdownProps = {
+	className?: string;
+	disabled?: boolean;
+	variant?: string;
+	children: JSX.Element[];
+};
+
+export const SplitDropdown: React.FC< SplitDropdownProps > = ( {
+	className = '',
+	children,
+	disabled,
+	variant = 'primary',
+}: SplitDropdownProps ) => {
+	const groupActionProps = Object.assign(
+		{ variant },
+		disabled ? { disabled } : {}
+	);
+	const mainActionProps = {
+		...groupActionProps,
+		className: `woocommerce-split-dropdown__main-button ${ className }`,
+	};
+	const [ mainAction, ...menuActions ] = children;
```

Comment:
"Action" is not bad, but maybe this should be something generic like "item" given it doesn't necessarily have to be an action?

```suggestion
	const [ mainItem, ...menuItems ] = children;
```

Reply:
Yeah, that's better.